packages/@tinacms/toolkit/src/components/MutationSignal.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+
+Copyright 2021 Forestry.io Holdings, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import * as React from 'react'
+
+export const MutationSignalContext = React.createContext(-1)
+
+/**
+ *
+ */
+export const MutationSignalProvider = ({ children }: { children: any }) => {
+  const observerAreaRef = React.useRef<HTMLElement | null>(null)
+  const [signal, setSignal] = React.useState(0)
+  React.useEffect(() => {
+    if (!observerAreaRef) return
+
+    const observer = new MutationObserver(() => setSignal((s) => s + 1))
+    observer.observe(observerAreaRef.current as Node, {
+      childList: true,
+      subtree: true,
+      characterData: true,
+    })
+    return () => observer.disconnect()
+  }, [])
+  return (
+    <MutationSignalContext.Provider value={signal}>
+      <div ref={observerAreaRef as any}>{children}</div>
```

Comment:
I didn't expect us to be listening to everything from one observer like this, I was under the impression that each `data-tinaField` element would be `observed` individually rather than the entire DOM. What you have here seems simpler, but I wonder about performance. This does seem like the right place to start, though.

As I'm thinking about the more granular approach, it seems like it might be harder to know when a new `data-tinaField` element has entered the DOM, we'd have to have some sort of mechanism to trigger a DOM query, which then brings me back to the ref-based solution since it'd automatically register the DOM nodes as the enter the page. 🤔